// culqi-nodejs/dist/resources/v2/charges.resource.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Charges = void 0;
const base_resource_1 = require("../base.resource");
/**
 * Charges endpoints for API v2
 *
 * @see {@link https://apidocs.culqi.com/#tag/Cargos Documentation}.
 * */
class Charges extends base_resource_1.BaseResource {
    constructor(http, apiVersion) {
        super(http, `/v${apiVersion}/charges`);
    }
    /**
     * Create charge
     *
     * @see {@link https://github.com/iscodex/culqi-nodejs/blob/main/src/examples/charges/create.ts Usage Example}
     * */
    create(data) {
        return this.post(undefined, { data });
    }
    /**
     * Get charges
     *
     * @see {@link https://github.com/iscodex/culqi-nodejs/blob/main/src/examples/charges/find-by.ts Usage Example}
     * */
    findBy(params) {
        return this.get(undefined, { params });
    }
    /**
     * Get charge
     *
     * @see {@link https://github.com/iscodex/culqi-nodejs/blob/main/src/examples/charges/find.ts Usage Example}
     * */
    find(id) {
        return this.get(`/${id}`);
    }
    /**
     * Update charge metadata
     *
     * @see {@link https://github.com/iscodex/culqi-nodejs/blob/main/src/examples/charges/update.ts Usage Example}
     * */
    update(id, data) {
        return this.patch(`/${id}`, { data });
    }
    /**
     * Capture charge
     *
     * @see {@link https://github.com/iscodex/culqi-nodejs/blob/main/src/examples/charges/capture.ts Usage Example}
     * */
    capture(id) {
        return this.post(`/${id}/capture`);
    }
}
exports.Charges = Charges;
