// culqi-nodejs/dist/resources/v2/orders.resource.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Orders = void 0;
const base_resource_1 = require("../base.resource");
/**
 * Orders endpoints for API v2
 *
 * @see {@link https://apidocs.culqi.com/#tag/Ordenes Documentation}.
 * */
class Orders extends base_resource_1.BaseResource {
    constructor(http, apiVersion) {
        super(http, `/v${apiVersion}/orders`);
    }
    /**
     * Create order
     *
     * @see {@link https://github.com/iscodex/culqi-nodejs/blob/main/src/examples/orders/create.ts Usage Example}
     * */
    create(data) {
        return this.post(undefined, { data });
    }
    /**
     * Get orders
     *
     * @see {@link https://github.com/iscodex/culqi-nodejs/blob/main/src/examples/orders/find-by.ts Usage Example}
     * */
    findBy(params) {
        return this.get(undefined, { params });
    }
    /**
     * Confirm order
     *
     * @see {@link https://github.com/iscodex/culqi-nodejs/blob/main/src/examples/orders/confirm.ts Usage Example}
     * */
    confirm(id) {
        return this.post(`/${id}/confirm`, { pub: true });
    }
    /**
     * Confirm order type
     *
     * @see {@link https://github.com/iscodex/culqi-nodejs/blob/main/src/examples/orders/confirm-with-type.ts Usage Example}
     * */
    confirmWithType(data) {
        return this.post(`/confirm`, { data, pub: true });
    }
    /**
     * Get order
     *
     * @see {@link https://github.com/iscodex/culqi-nodejs/blob/main/src/examples/orders/find.ts Usage Example}
     * */
    find(id) {
        return this.get(`/${id}`);
    }
    /**
     * Update order
     *
     * @see {@link https://github.com/iscodex/culqi-nodejs/blob/main/src/examples/orders/update.ts Usage Example}
     * */
    update(id, data) {
        return this.patch(`/${id}`, { data });
    }
    /**
     * Delete order
     *
     * @see {@link https://github.com/iscodex/culqi-nodejs/blob/main/src/examples/orders/remove.ts Usage Example}
     * */
    remove(id) {
        return this.del(`/${id}`);
    }
}
exports.Orders = Orders;
