// culqi-nodejs/dist/resources/v2/subscriptions.resource.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Subscriptions = void 0;
const base_resource_1 = require("../base.resource");
/**
 * Subscriptions endpoints for API v2
 *
 * @see {@link https://apidocs.culqi.com/#tag/Tarjetas Documentation}.
 * */
class Subscriptions extends base_resource_1.BaseResource {
    constructor(http, apiVersion) {
        super(http, `/v${apiVersion}/recurrent/subscriptions`);
    }
    /**
     * Create subscription
     *
     * @see {@link https://github.com/iscodex/culqi-nodejs/blob/main/src/examples/subscriptions/create.ts Usage Example}
     * */
    create(data) {
        return this.post('/create', { data });
    }
    /**
     * Get subscriptions
     *
     * @see {@link https://github.com/iscodex/culqi-nodejs/blob/main/src/examples/subscriptions/find-by.ts Usage Example}
     * */
    findBy(params) {
        return this.get(undefined, { params });
    }
    /**
     * Get subscription
     *
     * @see {@link https://github.com/iscodex/culqi-nodejs/blob/main/src/examples/subscriptions/find.ts Usage Example}
     * */
    find(id) {
        return this.get(`/${id}`);
    }
    /**
     * Update subscription
     *
     * @see {@link https://github.com/iscodex/culqi-nodejs/blob/main/src/examples/subscriptions/update.ts Usage Example}
     * */
    update(id, data) {
        return this.patch(`/${id}`, { data });
    }
    /**
     * Cancel subscription
     *
     * @see {@link https://github.com/iscodex/culqi-nodejs/blob/main/src/examples/subscriptions/remove.ts Usage Example}
     * */
    remove(id) {
        return this.del(`/${id}`);
    }
}
exports.Subscriptions = Subscriptions;
