// culqi-nodejs/dist/resources/v2/tokens.resource.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Tokens = void 0;
const base_resource_1 = require("../base.resource");
/**
 * Tokens endpoints for API v2
 *
 * @see {@link https://apidocs.culqi.com/#tag/Tokens Documentation}.
 * */
class Tokens extends base_resource_1.BaseResource {
    constructor(http, apiVersion) {
        super(http, `/v${apiVersion}/tokens`);
    }
    /**
     * Create token
     *
     * @see {@link https://github.com/iscodex/culqi-nodejs/blob/main/src/examples/tokens/create.ts Usage Example}
     * */
    create(data) {
        return this.post(undefined, { data, pub: true });
    }
    /**
     * Get tokens
     *
     * @see {@link https://github.com/iscodex/culqi-nodejs/blob/main/src/examples/tokens/find-by.ts Usage Example}
     * */
    findBy(params) {
        return this.get(undefined, { params });
    }
    /**
     * Get token
     *
     * @see {@link https://github.com/iscodex/culqi-nodejs/blob/main/src/examples/tokens/find.ts Usage Example}
     * */
    find(id) {
        return this.get(`/${id}`);
    }
    /**
     * Update token metadata
     *
     * @see {@link https://github.com/iscodex/culqi-nodejs/blob/main/src/examples/tokens/update.ts Usage Example}
     * */
    update(id, data) {
        return this.patch(`/${id}`, { data });
    }
    /**
     * Create token (Yape)
     *
     * @see {@link https://github.com/iscodex/culqi-nodejs/blob/main/src/examples/tokens/create-yape.ts Usage Example}
     * */
    createYape(data) {
        return this.post('/yape', { data, pub: true });
    }
}
exports.Tokens = Tokens;
