// culqi-nodejs@1.1.0/dist/client/culqi-client.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CulqiClient = void 0;
const resource_factory_1 = require("../utils/resource-factory");
const http_client_1 = require("./http-client");
/** High-level Culqi client exposing versioned resources. */
class CulqiClient {
    opts;
    // Typed resources (different per version)
    tokens;
    charges;
    refunds;
    customers;
    cards;
    plans;
    subscriptions;
    orders;
    events;
    statements;
    constructor(opts) {
        this.opts = opts;
        const http = new http_client_1.HttpClient(opts);
        const r = (0, resource_factory_1.buildResources)(opts.apiVersion, http);
        this.tokens = r.tokens;
        this.charges = r.charges;
        this.refunds = r.refunds;
        this.customers = r.customers;
        this.cards = r.cards;
        this.plans = r.plans;
        this.subscriptions = r.subscriptions;
        this.orders = r.orders;
        this.events = r.events;
        this.statements = r.statements;
    }
    /** Fluent initializer with sensible defaults. */
    static init(opts) {
        return new CulqiClient({
            baseUrl: 'https://api.culqi.com',
            retries: 2,
            timeout: 8000,
            ...opts,
        });
    }
}
exports.CulqiClient = CulqiClient;
