// culqi-nodejs@1.1.0/dist/resources/v2/cards.resource.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Cards = void 0;
const base_resource_1 = require("../base.resource");
/**
 * Cards endpoints for API v2
 *
 * @see {@link https://apidocs.culqi.com/#tag/Tarjetas Documentation}.
 * */
class Cards extends base_resource_1.BaseResource {
    constructor(http, apiVersion) {
        super(http, `/v${apiVersion}/cards`);
    }
    /**
     * Create card
     *
     * @see {@link https://github.com/iscodex/culqi-nodejs/blob/main/src/examples/cards/create.ts Usage Example}
     * */
    create(data) {
        return this.post(undefined, { data });
    }
    /**
     * Get cards
     *
     * @see {@link https://github.com/iscodex/culqi-nodejs/blob/main/src/examples/cards/find-by.ts Usage Example}
     * */
    findBy(params) {
        return this.get(undefined, { params });
    }
    /**
     * Get card
     *
     * @see {@link https://github.com/iscodex/culqi-nodejs/blob/main/src/examples/cards/find.ts Usage Example}
     * */
    find(id) {
        return this.get(`/${id}`);
    }
    /**
     * Update card
     *
     * @see {@link https://github.com/iscodex/culqi-nodejs/blob/main/src/examples/cards/update.ts Usage Example}
     * */
    update(id, data) {
        return this.patch(`/${id}`, { data });
    }
    /**
     * Delete card
     *
     * @see {@link https://github.com/iscodex/culqi-nodejs/blob/main/src/examples/cards/remove.ts Usage Example}
     * */
    remove(id) {
        return this.del(`/${id}`);
    }
}
exports.Cards = Cards;
