// culqi-nodejs@1.1.0/dist/resources/v2/plans.resource.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Plans = void 0;
const base_resource_1 = require("../base.resource");
/**
 * Plans endpoints for API v2
 *
 * @see {@link https://apidocs.culqi.com/#tag/Planes Documentation}.
 * */
class Plans extends base_resource_1.BaseResource {
    constructor(http, apiVersion) {
        super(http, `/v${apiVersion}/recurrent/plans`);
    }
    /**
     * Create plan
     *
     * @see {@link https://github.com/iscodex/culqi-nodejs/blob/main/src/examples/plans/create.ts Usage Example}
     * */
    create(data) {
        return this.post('/create', { data });
    }
    /**
     * Get plans
     *
     * @see {@link https://github.com/iscodex/culqi-nodejs/blob/main/src/examples/plans/find-by.ts Usage Example}
     * */
    findBy(params) {
        return this.get(undefined, { params });
    }
    /**
     * Get plan
     *
     * @see {@link https://github.com/iscodex/culqi-nodejs/blob/main/src/examples/plans/find.ts Usage Example}
     * */
    find(id) {
        return this.get(`/${id}`);
    }
    /**
     * Update plan
     *
     * @see {@link https://github.com/iscodex/culqi-nodejs/blob/main/src/examples/plans/update.ts Usage Example}
     * */
    update(id, data) {
        return this.patch(`/${id}`, { data });
    }
    /**
     * Delete plan
     *
     * @see {@link https://github.com/iscodex/culqi-nodejs/blob/main/src/examples/plans/remove.ts Usage Example}
     * */
    remove(id) {
        return this.del(`/${id}`);
    }
}
exports.Plans = Plans;
