// culqi-nodejs@1.1.0/dist/resources/v2/statements.resource.spec.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const culqi_client_1 = require("../../client/culqi-client");
const http_client_1 = require("../../client/http-client");
let sdk;
beforeEach(() => {
    sdk = culqi_client_1.CulqiClient.init({
        publicKey: 'pk_test_x',
        secretKey: 'sk_test_x',
        apiVersion: '2',
    });
});
afterEach(() => jest.restoreAllMocks());
describe('Statements – Unit Test (v2)', () => {
    it('should get a list of operations with given query params', async () => {
        const postSpy = jest.spyOn(http_client_1.HttpClient.prototype, 'get').mockResolvedValue({ data: [] });
        const query = { currency_code: 'PEN', operation_type: 'Authorized' };
        await sdk.statements.operations(query);
        expect(postSpy).toHaveBeenCalledWith('/v2/balance/operations', {
            params: query,
            pub: false,
        });
    });
    it('should get a list of deposits with given query params', async () => {
        const postSpy = jest.spyOn(http_client_1.HttpClient.prototype, 'get').mockResolvedValue({ data: [] });
        const query = {
            currency_code: 'PEN',
            date_from: 1476132639,
            date_to: 1476132639,
        };
        await sdk.statements.deposits(query);
        expect(postSpy).toHaveBeenCalledWith('/v2/balance/deposits', {
            params: query,
            pub: false,
        });
    });
    it('should get a single deposit by its id', async () => {
        const postSpy = jest
            .spyOn(http_client_1.HttpClient.prototype, 'get')
            .mockResolvedValue({ id: 'fARY6gP7VeRiI2L2' });
        const res = await sdk.statements.deposit('fARY6gP7VeRiI2L2');
        expect(postSpy).toHaveBeenCalledWith('/v2/balance/deposits/fARY6gP7VeRiI2L2', { pub: false });
        expect(res).toEqual({ id: 'fARY6gP7VeRiI2L2' });
        postSpy.mockRestore();
    });
    it('should get a list of billings with given query params', async () => {
        const postSpy = jest.spyOn(http_client_1.HttpClient.prototype, 'get').mockResolvedValue({ data: [] });
        const query = {
            currency_code: 'PEN',
            operation_type: 'Authorized',
        };
        await sdk.statements.billings(query);
        expect(postSpy).toHaveBeenCalledWith('/v2/balance/billing-operations', {
            params: query,
            pub: false,
        });
    });
});
