// culqi-nodejs@1.1.0/dist/resources/v2/subscriptions.resource.spec.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const culqi_client_1 = require("../../client/culqi-client");
const http_client_1 = require("../../client/http-client");
let sdk;
beforeEach(() => {
    sdk = culqi_client_1.CulqiClient.init({
        publicKey: 'pk_test_x',
        secretKey: 'sk_test_x',
        apiVersion: '2',
    });
});
afterEach(() => jest.restoreAllMocks());
describe('Subscriptions – Unit Test (v2)', () => {
    it('should create a subscription', async () => {
        const postSpy = jest.spyOn(http_client_1.HttpClient.prototype, 'post').mockResolvedValue({ id: 'sxn_123' });
        const payload = {
            card_id: 'crd_123',
            plan_id: 'sxn_123',
            tyc: true,
        };
        const res = await sdk.subscriptions.create(payload);
        expect(postSpy).toHaveBeenCalledWith('/v2/recurrent/subscriptions/create', {
            data: payload,
            pub: false,
        });
        expect(res).toEqual({ id: 'sxn_123' });
        postSpy.mockRestore();
    });
    it('should get a list of subscriptions with given query params', async () => {
        const postSpy = jest.spyOn(http_client_1.HttpClient.prototype, 'get').mockResolvedValue({ data: [] });
        const query = { limit: 10, status: 2 };
        await sdk.subscriptions.findBy(query);
        expect(postSpy).toHaveBeenCalledWith('/v2/recurrent/subscriptions', {
            params: query,
            pub: false,
        });
    });
    it('should get a single subscription by its id', async () => {
        const postSpy = jest.spyOn(http_client_1.HttpClient.prototype, 'get').mockResolvedValue({ id: 'sxn_123' });
        const res = await sdk.subscriptions.find('sxn_123');
        expect(postSpy).toHaveBeenCalledWith('/v2/recurrent/subscriptions/sxn_123', { pub: false });
        expect(res).toEqual({ id: 'sxn_123' });
        postSpy.mockRestore();
    });
    it('should update a subscription', async () => {
        const postSpy = jest.spyOn(http_client_1.HttpClient.prototype, 'patch').mockResolvedValue({ id: 'sxn_123' });
        const payload = {
            card_id: 'crd_123',
            metadata: { foo: 'bar' },
        };
        await sdk.subscriptions.update('sxn_123', payload);
        expect(postSpy).toHaveBeenCalledWith('/v2/recurrent/subscriptions/sxn_123', {
            data: payload,
            pub: false,
        });
    });
    it('should delete a subscription', async () => {
        const postSpy = jest
            .spyOn(http_client_1.HttpClient.prototype, 'del')
            .mockResolvedValue({ id: 'sxn_123', deleted: true });
        const res = await sdk.subscriptions.remove('sxn_123');
        expect(postSpy).toHaveBeenCalledWith('/v2/recurrent/subscriptions/sxn_123', { pub: false });
        expect(res).toEqual({ id: 'sxn_123', deleted: true });
        postSpy.mockRestore();
    });
});
