// culqi-nodejs/dist/utils/resource-factory.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.buildResources = buildResources;
const V2 = __importStar(require("../resources/v2"));
// import * as V3 from '../resources/v3';
/**
 * Returns an object with the appropriate resources for the given API version.
 */
function buildResources(version, http) {
    switch (version) {
        case '2':
            return {
                tokens: new V2.Tokens(http, version),
                charges: new V2.Charges(http, version),
                refunds: new V2.Refunds(http, version),
                customers: new V2.Customers(http, version),
                cards: new V2.Cards(http, version),
                plans: new V2.Plans(http, version),
                subscriptions: new V2.Subscriptions(http, version),
                orders: new V2.Orders(http, version),
                events: new V2.Events(http, version),
                statements: new V2.Statements(http, version),
            };
        default:
            throw new Error(`Unsupported API version: ${version}`);
    }
}
