// culqi-nodejs/dist/resources/v2/cards.resource.spec.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const culqi_client_1 = require("../../client/culqi-client");
const http_client_1 = require("../../client/http-client");
let sdk;
beforeEach(() => {
    sdk = culqi_client_1.CulqiClient.init({
        publicKey: 'pk_test_x',
        secretKey: 'sk_test_x',
        apiVersion: '2',
    });
});
afterEach(() => jest.restoreAllMocks());
describe('Cards - Unit Test (v2)', () => {
    it('should create a card', async () => {
        const postSpy = jest.spyOn(http_client_1.HttpClient.prototype, 'post').mockResolvedValue({ id: 'crd_123' });
        const payload = {
            customer_id: 'cus_123',
            token_id: 'tkn_123',
            validate: true,
        };
        const res = await sdk.cards.create(payload);
        expect(postSpy).toHaveBeenCalledWith('/v2/cards', {
            data: payload,
            pub: false,
        });
        expect(res).toEqual({ id: 'crd_123' });
        postSpy.mockRestore();
    });
    it('should get a list of cards with given query params', async () => {
        const postSpy = jest.spyOn(http_client_1.HttpClient.prototype, 'get').mockResolvedValue({ data: [] });
        const query = { limit: 10, creation_date: 1476132639 };
        await sdk.cards.findBy(query);
        expect(postSpy).toHaveBeenCalledWith('/v2/cards', {
            params: query,
            pub: false,
        });
    });
    it('should get a single card by its id', async () => {
        const postSpy = jest.spyOn(http_client_1.HttpClient.prototype, 'get').mockResolvedValue({ id: 'crd_123' });
        const res = await sdk.cards.find('crd_123');
        expect(postSpy).toHaveBeenCalledWith('/v2/cards/crd_123', { pub: false });
        expect(res).toEqual({ id: 'crd_123' });
        postSpy.mockRestore();
    });
    it('should update a card', async () => {
        const postSpy = jest.spyOn(http_client_1.HttpClient.prototype, 'patch').mockResolvedValue({ id: 'crd_123' });
        const payload = { token_id: 'tkn_123', metadata: { foo: 'bar' } };
        await sdk.cards.update('crd_123', payload);
        expect(postSpy).toHaveBeenCalledWith('/v2/cards/crd_123', {
            data: payload,
            pub: false,
        });
    });
});
