// culqi-nodejs/dist/resources/v2/customers.resource.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Customers = void 0;
const base_resource_1 = require("../base.resource");
/**
 * Customers endpoints for API v2
 *
 * @see {@link https://apidocs.culqi.com/#tag/Clientes Documentation}.
 * */
class Customers extends base_resource_1.BaseResource {
    constructor(http, apiVersion) {
        super(http, `/v${apiVersion}/customers`);
    }
    /**
     * Create customer
     *
     * @see {@link https://github.com/iscodex/culqi-nodejs/blob/main/src/examples/customers/create.ts Usage Example}
     * */
    create(data) {
        return this.post(undefined, { data });
    }
    /**
     * Get customers
     *
     * @see {@link https://github.com/iscodex/culqi-nodejs/blob/main/src/examples/customers/find-by.ts Usage Example}
     * */
    findBy(params) {
        return this.get(undefined, { params });
    }
    /**
     * Get customer
     *
     * @see {@link https://github.com/iscodex/culqi-nodejs/blob/main/src/examples/customers/find.ts Usage Example}
     * */
    find(id) {
        return this.get(`/${id}`);
    }
    /**
     * Update customer
     *
     * @see {@link https://github.com/iscodex/culqi-nodejs/blob/main/src/examples/customers/update.ts Usage Example}
     * */
    update(id, data) {
        return this.patch(`/${id}`, { data });
    }
    /**
     * Delete customer
     *
     * @see {@link https://github.com/iscodex/culqi-nodejs/blob/main/src/examples/customers/remove.ts Usage Example}
     * */
    remove(id) {
        return this.del(`/${id}`);
    }
}
exports.Customers = Customers;
