// culqi-nodejs/dist/resources/v2/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Statements = exports.Events = exports.Orders = exports.Subscriptions = exports.Plans = exports.Cards = exports.Customers = exports.Refunds = exports.Charges = exports.Tokens = void 0;
var tokens_resource_1 = require("./tokens.resource");
Object.defineProperty(exports, "Tokens", { enumerable: true, get: function () { return tokens_resource_1.Tokens; } });
var charges_resource_1 = require("./charges.resource");
Object.defineProperty(exports, "Charges", { enumerable: true, get: function () { return charges_resource_1.Charges; } });
var refunds_resource_1 = require("./refunds.resource");
Object.defineProperty(exports, "Refunds", { enumerable: true, get: function () { return refunds_resource_1.Refunds; } });
var customers_resource_1 = require("./customers.resource");
Object.defineProperty(exports, "Customers", { enumerable: true, get: function () { return customers_resource_1.Customers; } });
var cards_resource_1 = require("./cards.resource");
Object.defineProperty(exports, "Cards", { enumerable: true, get: function () { return cards_resource_1.Cards; } });
var plans_resource_1 = require("./plans.resource");
Object.defineProperty(exports, "Plans", { enumerable: true, get: function () { return plans_resource_1.Plans; } });
var subscriptions_resource_1 = require("./subscriptions.resource");
Object.defineProperty(exports, "Subscriptions", { enumerable: true, get: function () { return subscriptions_resource_1.Subscriptions; } });
var orders_resource_1 = require("./orders.resource");
Object.defineProperty(exports, "Orders", { enumerable: true, get: function () { return orders_resource_1.Orders; } });
var events_resource_1 = require("./events.resource");
Object.defineProperty(exports, "Events", { enumerable: true, get: function () { return events_resource_1.Events; } });
var statements_resource_1 = require("./statements.resource");
Object.defineProperty(exports, "Statements", { enumerable: true, get: function () { return statements_resource_1.Statements; } });
